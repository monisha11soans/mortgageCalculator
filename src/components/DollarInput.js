import React from "react";
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import InputAdornment from '@material-ui/core/InputAdornment';
import OutlinedInput from '@material-ui/core/OutlinedInput';

export const DollarInput = ({ id, labelName, targetValue, handleChange, disabledVal}) => {
    return (
        <FormControl className="mortgageItem" style={{ "width": "350px", "margin": "8px" }}
            size="small" variant="outlined">
            <InputLabel htmlFor={id}>{labelName}</InputLabel>
            <OutlinedInput
                id={id}
                value={targetValue}
                onChange= {handleChange}
                startAdornment={<InputAdornment position="start">$</InputAdornment>}
                labelWidth={100}
                size="small"
                disabled={disabledVal}
            />
        </FormControl>
    )
}