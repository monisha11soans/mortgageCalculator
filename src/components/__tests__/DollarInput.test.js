import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import React from 'react';
import { DollarInput } from '../DollarInput';

configure({ adapter: new Adapter() });

describe('Mortgage calculator Component', () => {
    const handleFunction = jest.fn();

    const props = {
        id: 'housePrice',
        labelName: 'Home Price',
        targetValue: 500,
        handleChange: handleFunction,
        disabledVal: false
    };

    const tree = renderer.create(<DollarInput {...props} />).toJSON();

    it('should render properly', () => {
        expect(tree).toMatchSnapshot();
    });
});