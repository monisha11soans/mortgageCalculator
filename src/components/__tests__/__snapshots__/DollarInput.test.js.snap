// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mortgage calculator Component should render properly 1`] = `
<div
  className="MuiFormControl-root mortgageItem"
  style={
    Object {
      "margin": "8px",
      "width": "350px",
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="housePrice"
  >
    Home Price
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    onClick={[Function]}
    size="small"
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        $
      </p>
    </div>
    <input
      aria-invalid={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
      disabled={false}
      id="housePrice"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value={500}
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-2"
        style={
          Object {
            "width": 83,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
